searchcore/src/vespa/searchcore/fdispatch/search/mergehits.cpp

```diff
@@ -65,10 +67,16 @@ FastS_MergeCopyHit(typename T::HitType *src,
     dst->setDistributionKey(src->getDistributionKey());
 }
 
+struct GlobalIdHasher {
+    vespalib::hash_set<document::GlobalId, document::GlobalId::hash> seenSet;
```

Comment:
pre-size this one based on the expected number of entries (for example 3*expsize)